joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/dataflow/DataFlowTests.scala

```diff
@@ -1259,6 +1259,33 @@ class DataFlowTests extends RubyCode2CpgFixture(withPostProcessing = true, withD
     }
   }
 
+  "Data flow for begin/rescue with sink in function within do block" ignore {
+    val cpg = code("""
+        |def foo(arg)
+        |  puts "in begin"
+        |  arg do |y|
+        |  return y
+        |rescue SomeException
+        |  puts "Caught SomeException"
+        |rescue => exvar
+        |  puts "Caught exception in variable #{exvar}"
+        |rescue
+        |  puts "Catch-all block"
+        |end
+        |
+        |x = 1
+        |z = foo x
+        |puts z
+        |
+        |""".stripMargin)
+
+    "find flows to the sink" in {
+      val source = cpg.identifier.name("x").l
+      val sink   = cpg.call.name("puts").l
```

Comment:
Better to specify which sink are you intending Ex- `cpg.calll.name("puts").lineNumber(9)`